### PR TITLE
Migrate query-builder from grid-styled

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/20163

### CODE REVIEW

**Comment by ranquild** [comment]
@kulyk @ariya I've manually tested the changes and everything seems to look the same, added you as reviewers to double check.

**Review by gusaiani — APPROVED** [review]
Markup is so much clearer, thank you Alexandr.

File: frontend/src/metabase/query_builder/components/QueryDownloadWidget.jsx [hunk 1]
```diff
@@ -40,22 +45,21 @@ const QueryDownloadWidget = ({
     triggerClasses={cx(className, "text-brand-hover")}
     triggerClassesClose={classNameClose}
   >
-    <Box
-      p={2}
-      width={result.data && result.data.rows_truncated != null ? 300 : 260}
+    <WidgetRoot
+      isExpanded={result.data && result.data.rows_truncated != null ? 300 : 260}
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/QueryDownloadWidget.jsx:49]: `WidgetRoot` sets `width` depending on `isExpanded`, but the ternary passes a numeric value here, so it's going to be `true` all the time

**Review by kulyk — CHANGES_REQUESTED** [review]
Overall looks good, I noticed a UI regression on the join step

**See how it looks on `master`**
notebook-joins-master

**And that's how it looks in the branch**

notebook-joins-now

**Comment by kulyk** [comment]
Also, could you please look at [this regression](https://metaboat.slack.com/archives/C02H619CJ8K/p1643922564501759) Bruno has mentioned on Slack? Seems like the icon sizes are not identical

**Comment by kulyk** [comment]
@metabase-bot run visual tests

**Comment by ranquild** [comment]
@kulyk Fixed these 2 issues

**Review by kulyk — APPROVED** [review]

**Pull request merged**